step-controller/step-controller-remote-client/src/main/java/step/client/resources/RemoteResourceManager.java

```diff
@@ -136,12 +137,32 @@ public Resource createResource(String resourceType, InputStream resourceStream,
 
 	@Override
 	public Resource createResource(String resourceType, boolean isDirectory, InputStream resourceStream, String resourceFileName,
-								   boolean checkForDuplicates, ObjectEnricher objectEnricher) throws IOException, SimilarResourceExistingException {
+								   boolean checkForDuplicates, ObjectEnricher objectEnricher) {
 		StreamDataBodyPart bodyPart = new StreamDataBodyPart("file", resourceStream, resourceFileName);
+
+		// !!! in fact, 'checkForDuplicates' parameter is ignored, because the list of found duplicated resources (with the same hash sums)
+		// is located in ResourceUploadResponse.similarResources, but we ignore this list here and just take the uploaded resource

```

Comment:
The local implementation of the ResourceManager (ResourceManagerImpl) throws a SimilarResourceExistingException when a similar resource already exists. In my opinion this remote implementation of the ResourceManager should behave the same. Any reason for not throwing a SimilarResourceExistingException in this case here?